h/services/search_index/service.py

```diff
@@ -74,8 +77,34 @@ def delete_annotation_by_id(self, annotation_id, refresh=False):
 
         self._index_annotation_body(annotation_id, {"deleted": True}, refresh=refresh)
 
-    def _index_annotation_body(self, annotation_id, body, refresh, target_index=None):
+    def handle_annotation_event(self, event, synchronous=False):
+        """
+        Process an annotation event, taking appropriate action to the event.
+
+        This will attempt to fulfill the request synchronously if asked, or
+        fall back on a delayed celery task if not or if this fails.
+
+        :param event: AnnotationEvent object
+        :synchronous: Try synchronous indexing before async
+        """
+        if event.action in ["create", "update"]:
+            sync_handler, async_task = self.add_annotation_by_id, add_annotation
+        elif event.action == "delete":
+            sync_handler, async_task = self.delete_annotation_by_id, delete_annotation
+        else:
+            return False
+
+        if synchronous:
+            try:
+                return sync_handler(event.annotation_id)
+
+            except Exception as err:
+                report_exception(err)
```

Comment:
So if synchronous indexing is failing sometimes, we should be able to see that in Sentry. We'll also be able to see it indirectly in Papertrail and New Relic by just looking at how often the asynchronous indexing runs